src/main/java/org/broadleafcommerce/frameworkmapping/autoconfigure/FrameworkMappingAutoConfiguration.java

```diff
@@ -57,6 +58,7 @@ protected HandlerMethod handleNoMatch(Set<RequestMappingInfo> infos,
                         } catch (Exception e) {
                             throw new IllegalStateException(e);
                         }
+                        new UrlPathHelper().resolveAndCacheLookupPath(request);
```

Comment:
Looks like there's a default instance available that can be reused. Not sure if this was known.

```suggestion
                        UrlPathHelper.defaultInstance.resolveAndCacheLookupPath(request);
```

Reply:
What was the motivation for this change? It looks like when `AbstractHandlerMapping#getHandler()` is called, this `resolveAndCacheLookupPath` method is also called: https://github.com/spring-projects/spring-framework/blob/84045374744aee245d1ed717454cb91e5d1c1ad3/spring-webmvc/src/main/java/org/springframework/web/servlet/handler/AbstractHandlerMapping.java#L517-L520

Presumably, by the time `handleNoMatch()` is called, `getHandler()` should have already been called on `RequestMappingHandlerMapping`. Thus, this cache attribute should have already been set on the request.

Was this suddenly encountering some bug?

Reply:
Without it, it most definitely fails with a missing request attribute.

Reply:
Basically, a version of this: https://www.javacodegeeks.com/2022/04/fixing-the-spring-web-error-expected-lookuppath-in-request-attribute-org-springframework-web-util-urlpathhelper-path.html